recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,1418 @@
+        # but assume module is DISABLED if build_all_modules=True and ALL of the options_to_check are "NO"
+        #
+        # See notes above, VTK may do even deeper checks but this hopefully will be good enough.
+        #
+        if self.options.build_all_modules:
+            for option in options_to_check:
+                if not _is_module_NO(option):
+                    return True # something was not 'NO', then this module is at least partially enabled
+            # ALL are NO, then assume this module is disabled
+            return False
+        else:
+            for option in options_to_check:
+                if _is_module_YES_or_WANT(option):
+                    return True
+            return False
+
+
+    # special case for QT as it is more involved
+    @property
+    def _is_any_Qt_enabled(self):
+        return self._is_module_enabled([
+            self.options.group_enable_Qt,
+            self.options.module_enable_GUISupportQt,
+            self.options.module_enable_GUISupportQtQuick,
+            self.options.module_enable_GUISupportQtSQL,
+            self.options.module_enable_RenderingQt,
+            self.options.module_enable_ViewsQt
+            ])
+
+
+    # Required as VTK adds 'd' to the end of library files on windows
+    @property
+    def _lib_suffix(self):
+        return "d" if self.settings.os == "Windows" and self.settings.build_type == "Debug" else ""
+
+    def build_requirements(self):
+        self.tool_requires("sqlite3/3.41.1")
+
+    def source(self):
+        if self.options.use_source_from_git:
+            self.run("git clone -b release --single-branch " + self.git_url + " " + self.source_folder)
+            # note: we give the branch a name so we don't have detached heads
+            # TODO change to standard git and python chdir
+            # Version: support targeting a commit hash instead of a version tag, assume version number < 8 chars long
+            # 1.34.67 ... only 7 chars long
+            # anything more, assume a branch name or a hash (or short hash)
+            git_hash = self.version if len(self.version) > 7 else "v" + self.version
+            self.run("cd " + self.source_folder + " && git checkout -b branch-" + git_hash + " " + git_hash)
+        else:
+            get(self, **self.conan_data["sources"][self.version],
+                    strip_root=True,
+                    destination=self.source_folder)
+
+    def _third_party(self):
+        parties = {
+                # LEFT field:  target name for linking, will be used as TARGET::TARGET in package_info()
+                # RIGHT field: package/version to require
+                "cli11":             "cli11/2.3.2",
+                "double-conversion": "double-conversion/3.2.1",
+                "eigen":             "eigen/3.4.0",
+                "expat":             "expat/2.5.0",
+                "exprtk":            "exprtk/0.0.1",   # TODO upgrade to 0.0.2 (there was a problem with first attempt)
```

Comment:
this patch is also required for 0.0.2
https://gitlab.kitware.com/vtk/vtk/-/commit/db8971b583aa538b52b2d47f000115e8e1b10524.patch

Reply:
Thanks for all these, have you considered forking my recipe in your own github repo so I can diff and merge your changes?

Reply:
@paulharris I only need FiltersGeneral for my purposes so my recipe was created by removing lots of things from the recipe in this PR and hardcoding modules. I can push it to github but I don't think that it will be very helpful for making universal recipe.